### PR TITLE
Hdr parser

### PR DESCRIPTION
**Pull request by slimbuck**
Move the HDR texture parser from scripts into the engine.

This PR supersedes #2956.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -159,6 +159,7 @@ export { JsonHandler } from './resources/json.js';
 export { JsonStandardMaterialParser } from './resources/parser/material/json-standard-material.js';
 export { KtxParser } from './resources/parser/texture/ktx.js';
 export { LegacyDdsParser } from './resources/parser/texture/legacy-dds.js';
+export { HdrParser } from './resources/parser/texture/hdr.js';
```
Comment (willeastcott) [inline src/index.js:162]: Is it necessary to export this class to the `pc` namespace? (I know we already export the other parsers - just wondering...)

File: src/resources/parser/texture/hdr.js [hunk 2]
```diff
@@ -19,78 +29,80 @@ Object.assign(Reader.prototype, {
...
 /**
  * @class
  * @name HdrParser
  * @implements {TextureParser}
  * @classdesc Texture parser for hdr files.
  */
-function HdrParser(registry) {
-    this.maxRetries = 0;
-}
+class HdrParser {
```
Comment (willeastcott) [inline src/resources/parser/texture/hdr.js:59]: And if this _is_ made public with documentation, currently, docs are pretty bare. 😄

Comment (willeastcott) [reply src/resources/parser/texture/hdr.js:59]: Basically, consider whether this should possibly be private.

File: src/index.js (hunk 1 above)
Comment (slimbuck) [reply src/index.js:162]: I was wondering the same thing, I believe not. (I was surprised to see the others exported here).

Users can get to built-in parser instances via `app.loader.getHandler(pc.ASSET_TEXTURE).parsers.hdr`.

I vote to remove all the texture parsers from being exported?

File: src/resources/parser/texture/hdr.js (hunk 2 above)
Comment (slimbuck) [reply src/resources/parser/texture/hdr.js:59]: Private - and I'll do the same to the other parsers?

Comment (willeastcott) [reply src/resources/parser/texture/hdr.js:59]: Fine by me.

**Comment by slimbuck** [comment]
@willeastcott I've removed texture parsers from export and flagged the class docs as private 👍🏻

**Review by willeastcott — APPROVED** [review]

**Pull request merged**